src/NHibernate/Impl/FilterImpl.cs

```diff
@@ -93,10 +93,19 @@ public IFilter SetParameterList(string name, ICollection values)
 			if (values.Count > 0)
 			{
 				var e = values.GetEnumerator();
-				e.MoveNext();
-				if (!type.ReturnedClass.IsInstanceOfType(e.Current))
+				try
```

Comment:
I think we should modernize the whole method: cease taking a non generic collection parameter, but take a generic one.